src/emulator/functionsEmulatorRuntime.ts

```diff
@@ -545,13 +556,17 @@ async function ProcessHTTPS(frb: FunctionsRuntimeBundle, trigger: EmulatedTrigge
     };
 
     ephemeralServer.enable("trust proxy");
-    ephemeralServer.use(bodyParser.json({}));
-    ephemeralServer.use(bodyParser.text({}));
-    ephemeralServer.use(bodyParser.urlencoded({ extended: true }));
-    ephemeralServer.use(bodyParser.raw({ type: "*/*" }));
+    functionRouter.use(bodyParser.json({ verify: rawBodySaver }));
+    functionRouter.use(bodyParser.text({ verify: rawBodySaver }));
+    functionRouter.use(bodyParser.urlencoded({ extended: true, verify: rawBodySaver }));
+    functionRouter.use(bodyParser.raw({ type: "*/*", verify: rawBodySaver }));
 
-    ephemeralServer.all("/*", handler);
+    functionRouter.all("*", handler);
 
+    ephemeralServer.use(
```

Comment:
I think the '/*' is better, since it guarantees we don't drop anyone's express subpaths.  This server is short-lived and created by us so we don't need to worry about path abuse.

Reply:
False (I think?). I believe the actual functionality in production is that a request to `blah.cloudfunctions.net/myFunc` should work. I believe including the `/` will break this, but we should verify, Also I have no idea what you mean by path abuse.

Reply:
I mean since we are the only caller we don't need to worry about anyone else hitting this server at a path we can't support.  So supporting `/*` is not opening ourselves up to any trouble.

Reply:
Just to be clear, in prod these all route to the same function:
```
blah.cloudfunctions.net/myFunc
blah.cloudfunctions.net/myFunc/foo
blah.cloudfunctions.net/myFunc/foo/bar
```

They all go to `myFunc`.

Reply:
I just added [some tests](https://github.com/firebase/firebase-tools/pull/1366/commits/94cf934fcb46be207789a383b2db04d50a6ee534) to describe this, can you verify that these are(n't) correct?

Reply:
Commented over there.